janus/server/api.py

```diff
@@ -21,9 +24,11 @@
 )
 def collect_qc(collect_request: CollectQCRequest = Body(...)) -> CollectQCResponse | JSONResponse:
     """Collect qc metrics for the external request."""
-    service = CollectQCService(collect_request)
     try:
-        collected_qc_metrics: CollectQCResponse = service.collect_qc_metrics_for_request()
-        return collected_qc_metrics
+        workflow_service: WorkflowCollectQCService = get_workflow_service(collect_request.workflow)
+        collect_service = CollectQCService(
+            collect_qc_request=collect_request, collect_qc_service=workflow_service
+        )
```

Comment:
Is it necessary to instantiate the `CollectQCService`? It definitely should not accept the `collect_qc_request` in the constructor at least! Since it is a service providing some general functionality, it does not make sense to require it to accept a request when instantiating it.

I kind of like the idea of setting the collect qc service strategy on the collect qc service using a setter.